drivers/soundwire/cadence_master.c

```diff
@@ -1184,6 +1184,7 @@ int sdw_cdns_probe(struct sdw_cdns *cdns)
 	init_completion(&cdns->tx_complete);
 	cdns->bus.port_ops = &cdns_port_ops;
 
+	INIT_WORK(&cdns->work, cdns_update_slave_status_work);
```

Comment:
question: is this good or bad to have multiple work queues, one per link?

On one hand, I can see that this can help deal with multiple status changes in parallel on all 4 links.
On the other hand, could this lead to serialization issues?

Just thinking aloud before my morning coffee...

Reply:
In theory, it is independent for each link. I believe that it will be better to have multiple work queues will try it today.